### DISCUSSION TITLE
Deploy apps to external  kubernetes

### DISCUSSION THREAD

**Discussion by gun2rin** [apache/camel-karavan#1072] [Q&A] [answered]
Hi guys. We have karavan in some environment (ENV 1). For example launched in kuberless mode, but nevermind. We have an app.  We deployed an app to external registry by karavan's build button. It's ok. So now we need to deploy and start an app in external kubernetes (ENV 2).   What are the correct steps to do this?  
Do we need to add some kubernetes deploy script to the app? Then how this script would be started?
We'd be very grateful for help and examples. 

UPD 1:
I've added kubernetes-build.sh to the project, but it doesn't seem to affect anything. The application continues to run in docker next to karavan when the deploy button is pressed. I need to set somewhere else  KUBERNETES_SERVICE_HOST variable, according to the code in ConfigService.java.

```java
public static boolean inKubernetes() {
        if (inKubernetes == null) {
            inKubernetes = Objects.nonNull(System.getenv("KUBERNETES_SERVICE_HOST"));
        }
        return inKubernetes;
    }
```
@mgubaidullin  help )

UPD 2:
Figured out that i can customize default docker-build.sh template and call cicd api or add kubernetes deploy right here to this script.

**Comment by davsclaus** [comment, marked as answer]
Yes you can customize those build scripts anyhow you like
